test/system/practical_support_behaviors_test.rb

```diff
@@ -105,6 +105,29 @@ class PracticalSupportBehaviorsTest < ApplicationSystemTestCase
       end
     end
   end
+
+  describe 'hiding practical support' do
+    before do
+      @patient.practical_supports.create support_type: 'lodging',
+                                         source: 'Other funds (see notes)',
+                                         created_by: @user
+      go_to_practical_support_tab
+    end
+
+    it 'toggles the showing of the practical support tab' do
+      assert has_text? /Practical Support/i
+      c = Config.find_or_create_by(config_key: 'hide_practical_support')
```

Comment:
My experience is that creating objects like this in the test block (instead of the setup block) makes it kind of tough to infer exactly what we're trying to test, and doing stuff in the database after having the UI loaded is a pattern that can cause a lot of unintended side effects due to the database being out of sync with the UI. What do you think about either turning this into a one-liner and move it to the `before` block (see suggestion) or do it all using selenium (see, for example, `updating a config - insurance` in `update_configs_test`)?

In short, I think we should pare this test down to:

before block:
* given that I have a patient
* and a config of hide_practical_support: no (at this point, we're done changing the state of the database, and ready to load the UI)
* and I visit the `edit_patient_path(@patient)`

test block
* there shouldn't be a link for practical support

Thoughts?

```suggestion
      Config.create config_key: 'hide_practical_support', config_value: { options: ['No'] }
```

Reply:
Totally agree that the pattern can be problematic, which is why I threw `page.refresh` calls in. I think you're right though that this is overkill.

Reply:
Broke it out into two separate it blocks - looks like it actually increased the run time of the test slightly*, but it should be easier for future peeps to reason with.


* this is to be expected since updating db and calling page.refresh is cheap relative to logging in a second time